code/target/src/main/java/com/adobe/marketing/mobile/target/TargetExtension.java

```diff
@@ -127,12 +134,15 @@ protected TargetExtension(final ExtensionApi extensionApi) {
      */
     @VisibleForTesting
     protected TargetExtension(final ExtensionApi extensionApi, final DeviceInforming deviceInfoService, final Networking networkService,
-                              final UIService uiService, final TargetState targetState, final TargetPreviewManager targetPreviewManager,
+                              final UIService uiService, final UriOpening uriService, final Context context,
+                              final TargetState targetState, final TargetPreviewManager targetPreviewManager,
                               final TargetRequestBuilder requestBuilder, final TargetResponseParser responseParser) {
```

Comment:
nit: We shall need to revisit the #args here as it seems beyond manageable, for example pass service provider instance instead of individual services - deviceInfoService, networkService, uiService, uriService.

Reply:
Refactored to use static mock for `ServiceProvider`